### PR TITLE
Extended information when printing the path for execution plans

### PR DESCRIPTION
**Pull request by jjfumero**
#### Description

Extended information when printing the path for execution plans

#### Problem description

n/ a.

#### Backend/s tested

Mark the backends affected by this PR.

- [X] OpenCL
- [X] PTX
- [X] SPIRV

#### OS tested

Mark the OS where this PR is tested.

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

```bash
$ tornado-test -V uk.ac.manchester.tornado.unittests.executor.TestExecutor#test05
tornado --jvm "-Xmx6g -Dtornado.recover.bailout=False -Dtornado.unittests.verbose=True "  -m  tornado.unittests/uk.ac.manchester.tornado.unittests.tools.TornadoTestRunner  --params "uk.ac.manchester.tornado.unittests.executor.TestExecutor#test05"
WARNING: Using incubator modules: jdk.incubator.vector

Root
Task info: s0.t0
	Backend           : OPENCL
	Device            : NVIDIA GeForce RTX 3070 CL_DEVICE_TYPE_GPU (available)
	Dims              : 1
	Global work offset: [0, 0, 0]
	Global work size  : [16, 1, 1]
	Local  work size  : null
	Number of workgroups  : null

Root
 -> withWarmUp 
 -> withDevice(OPENCL:NVIDIA GeForce RTX 3070)
 -> withGridScheduler(<s0.t0::GlobalWorkGroup=[16, 1, 1]::LocalWorkGroup=null>)
 -> withThreadInfo 
 -> withProfiler(SILENT) 
Root
 -> withWarmUp 
 -> withDevice(OPENCL:NVIDIA GeForce RTX 3070)
 -> withGridScheduler(<s0.t0::GlobalWorkGroup=[16, 1, 1]::LocalWorkGroup=null>)
 -> withThreadInfo 
 -> withProfiler(SILENT) 
After the execution
Root
Test: class uk.ac.manchester.tornado.unittests.executor.TestExecutor#test05
	Running test: test05                     ................  [PASS] 
Test ran: 1, Failed: 0, Unsupported: 0
```

### CODE REVIEW

File: tornado-api/src/main/java/uk/ac/manchester/tornado/api/plan/types/WithGridScheduler.java [hunk 1]
```diff
@@ -18,16 +18,20 @@
...
+
+    public WithGridScheduler(TornadoExecutionPlan parent, GridScheduler gridScheduler) {
         super(parent);
+        this.gridScheduler = gridScheduler;
     }
 
     @Override
     public String toString() {
-        return parentLink.toString() + "\n -> withGridScheduler ";
+        return parentLink.toString() + "\n -> withGridScheduler(<" + gridScheduler + ">)";
```
Comment (stratika) [inline tornado-api/src/main/java/uk/ac/manchester/tornado/api/plan/types/WithGridScheduler.java:35]: I guess we need `gridScheduler.toString()` right?

**Review by stratika — COMMENTED** [review]
why do we have a new empty file (ExecutionPlanType.java)?

File: tornado-api/src/main/java/uk/ac/manchester/tornado/api/plan/types/WithGridScheduler.java (hunk 1 above)
Comment (jjfumero) [reply tornado-api/src/main/java/uk/ac/manchester/tornado/api/plan/types/WithGridScheduler.java:35]: no need, we override the `toString` method for the class `GridScheduler`

**Comment by jjfumero** [comment]
> why do we have a new empty file (ExecutionPlanType.java)?

What do you mean by empty? It is an abstract class that serves as a bridge for all specialised actions from an ExecutionPlan.

**Comment by mairooni** [comment]
> > why do we have a new empty file (ExecutionPlanType.java)?
> 
> What do you mean by empty? It is an abstract class that serves as a bridge for all specialised actions from an ExecutionPlan.

In the diff this appears as an empty file (no class declaration)

**Comment by jjfumero** [comment]
> > > why do we have a new empty file (ExecutionPlanType.java)?
> > 
> > 
> > What do you mean by empty? It is an abstract class that serves as a bridge for all specialised actions from an ExecutionPlan.
> 
> In the diff this appears as an empty file (no class declaration)

Got it. This was a duplicated file, left over from a prev. refactoring. It is now fixed.

**Review by mairooni — APPROVED** [review]

**Review by stratika — APPROVED** [review]

**Pull request merged**